webapp/src/pages/registrationAdmin/components/RegistrationAdminComponent.js

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from "react";
+import { withRouter } from "react-router";
+import { createColClassName } from "../../../utils/styling/styling";
+import "react-table/react-table.css";
+import ReactTable from 'react-table';
+import FormTextBox from "../../../components/form/FormTextBox";
+import { registrationAdminService } from "../../../services/registration/registration.admin.service";
+
+const DEFAULT_EVENT_ID = process.env.REACT_APP_DEFAULT_EVENT_ID || 1;
+
+class RegistrationAdminComponent extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        isLoading: true,
+        unconfirmedList: [],
+        filteredList: [],
+        error: "",
+        searchTerm:"",
+        loading: false,
+        confirming: false,
+        confirmError: ""
+      };
+    }
+
+    getUnconfirmed = () => {
+      registrationAdminService.getUnconfirmed(DEFAULT_EVENT_ID).then(response=> {
+        this.setState({
+          unconfirmedList: response.data,
+          error: response.error,
+          loading: false
+        }, () => this.filterList());
+      })
+    }
+
+    onConfirm = registrationId => {
+      this.setState({confirming: true}, ()=> {
+        registrationAdminService.confirm(registrationId).then(response=>{
+          this.setState({
+            confirmed: response.data,
+            confirmError: response.error,
+            confirming: false
+          }, ()=> {
+            this.getUnconfirmed();
+          });
+        });
+      });
+    }
+
+    onSearchChange = field => {
+      let value = field.target.value.toLowerCase();
+      this.setState({
+        searchTerm: value
+      }, ()=>this.filterList());      
+    }
+
+    filterList = () => {
+      let value = this.state.searchTerm;
+      let filteredList = this.state.unconfirmedList.filter(
+        u => u.firstname.toLowerCase().indexOf(value) > -1 || u.lastname.toLowerCase().indexOf(value) > -1 || u.email.toLowerCase().indexOf(value) > -1);
```

Comment:
Might need to rework this a little. If for example, your are searching for someone with the surname `John`, and you have people with the name `John`, then the filter will not include people with the surname `John`. You might need to filter on each field independently.

Reply:
@KaleabTessera not sure I follow? There is a single search term that is applied as an OR to all 3 fields of each record - so if you search for John it will include people with the name OR surname "John"  ?

Reply:
@avishkar58 So if there is match for people with the name John, then it won't filter people with the surname John, because the `or` condition will have already been met. This means the filtered list will only filter out people with the name John and won't even look at the surname.

Does that make sense?

Reply:
This is applied per record and not a search for a list of records so it is not a prob.